bitnami/zookeeper/templates/statefulset.yaml

```diff
@@ -250,13 +250,17 @@ spec:
           {{- if .Values.livenessProbe.enabled }}
           livenessProbe:
             exec:
-              {{- if not .Values.service.tls.disable_base_client_port }}
-              command: ['/bin/bash', '-c', 'echo "ruok" | timeout {{ .Values.livenessProbe.probeCommandTimeout }} nc -w {{ .Values.livenessProbe.probeCommandTimeout }} localhost {{ .Values.service.port }} | grep imok']
+              {{- if .Values.livenessProbe.useAdmin }}
+              command: ['/bin/bash', '-c', 'curl -s -m {{ .Values.livenessProbe.probeCommandTimeout }} http://localhost:8080/commands/ruok']
```

Comment:
This changes the behavior of the liveness command. Previously liveness and readiness were measured the same - whether the server responds to client traffic. There are times in the normal lifecycle of a ZooKeeper ensemble where a healthy server will not be responsive to clients (e.g. during a leader election). The new form of the check will ensure a positive liveness result during such times and a negative readiness result.